### DISCUSSION TITLE
Validate localstorage on load and reset if inconsistencies

### DISCUSSION THREAD

**Discussion by jlplenio** [pmndrs/zustand#2528] [Q&A]
I persist my store, which uses an interface, in localstorage, e.g. 
`export interface UserAddressTyp {
	name: string;
	street: string;
}`
Now the user manually manipulates their localstorage and edits the key `name` to `nam3`. This is accepted by zustand, eventhough it does not adhere to the interface and crashes my app, which expects `name` to exist. For me, it would be sufficient to reset the store and localstorage, if the load encounters inconsistencies as described.

**Comment by dai-shi** [comment]
By default, the storage implementation has no validation and no type checking. It's not really for production in that sense. (We should clarify that in docs. cc @dbritto-dev )
The solution is either create your own storage with validation or use `reviver` option in createJSONStorage as a hack/easy solution.

https://github.com/pmndrs/zustand/blob/e74fdc8bc9fcb81a6ec1b82affc4bfdd82889145/src/middleware/persist.ts#L27
